app/src/app.vue

```diff
@@ -85,6 +86,41 @@ export default defineComponent({
 			{ immediate: true }
 		);
 
+		async function setAppLanguage() {
+			let lang = 'en-US'; // fallback language, if no language is set
+			if (userStore.currentUser && userStore.currentUser.language) {
+				// if user has set it's own language, prefer user language
+				lang = userStore.currentUser.language;
+			} else if (serverStore.info?.project?.project_language) {
+				// otherwise use project language
+				lang = serverStore.info.project.project_language;
+			}
+			await setLanguage(lang);
+		}
+
+		watch(
+			() => appStore.hydrating,
+			async () => {
+				// initially set app language after hydration
+				if (!appStore.hydrating) await setAppLanguage();
+			}
+		);
+
+		watch(
+			() => userStore.currentUser?.language,
+			async (userLanguage) => {
+				// set project language if user's language was reset to project language (null)
+				if (!appStore.hydrating && !userLanguage) await setAppLanguage();
+			}
+		);
```

Comment:
This also happens in the hydration for the usersstore if i'm not mistaken. This might be double

Reply:
This is needed when the user's language is reset to Default Language on the user's detail page. However, you are right - it would be better to handle it there. I will do so.